### PR TITLE
C833 delete users with shared entities

### PR DESCRIPTION
**Pull request by javitonino**
Please CR @juanignaciosl

### CODE REVIEW

File: app/controllers/superadmin/users_controller.rb [hunk 1]
```diff
@@ -56,6 +56,7 @@ def update
   end
 
   def destroy
+    @user.shared_entities.map(&:entity).each(&:delete) if params[:destroy_shared]
```
Comment (juanignaciosl) [inline app/controllers/superadmin/users_controller.rb]: Wasn't this going to be only for organization deletion?

Comment (juanignaciosl) [inline app/controllers/superadmin/users_controller.rb]: Oh, I see that you've solved this at Central. I thought that you'd choose to delete shared entities on organization deletion, so the next user deletions just will work.

Comment (javitonino) [inline app/controllers/superadmin/users_controller.rb]: Yes, but the problem is that the first step of deleting an organization is deleting its users. Check the central PR to see when is `destroy_shared` set to `true`.

**Comment by javitonino** [comment]
Don't trust GH display of tests, they are passing if you click through. Please CR @gfiorav

File: app/models/user/db_service.rb [hunk 2]
```diff
@@ -349,14 +349,16 @@ def drop_user(conn = nil, username = nil)
...
               drop_owned_by_user(conn, username)
+
               conflict_database_conn = @user.in_database(
                 as: :cluster_admin,
                 'database' => database_with_conflicts
               )
               drop_owned_by_user(conflict_database_conn, username)
               ['cdb', 'cdb_importer', 'cartodb', 'public', @user.database_schema].each do |s|
-                drop_users_privileges_in_schema(s, [username])
+                drop_users_privileges_in_schema(s, [username], conn: conflict_database_conn)
```
Comment (gfiorav) [inline app/models/user/db_service.rb]: Explain this change please!

Comment (javitonino) [inline app/models/user/db_service.rb]: The problem here is that `drop_users_privileges_in_schema` opened a new DB connection to the user DB that was not closed. So after deleting the user, that connections was left in some strange case that prevented DB deletion but could not be closed with `pg_terminate_backend` :angry: . So database deletion failed.

Now I just reuse a single connection and explicitly close it just below this.

**Comment by gfiorav** [comment]
📦  it!

**Pull request merged**